ibm/service/cos/data_source_ibm_cos_bucket.go

```diff
@@ -638,6 +645,21 @@ func dataSourceIBMCosBucketRead(d *schema.ResourceData, meta interface{}) error
 			d.Set("replication_rule", replicationRules)
 		}
 	}
+	//object lock configuration
+	getObjectLockConfigurationInput := &s3.GetObjectLockConfigurationInput{
+		Bucket: aws.String(bucketName),
+	}
+	output, err := s3Client.GetObjectLockConfiguration(getObjectLockConfigurationInput)
+	if err != nil && !strings.Contains(err.Error(), "AccessDenied: Access Denied") {
+		return err
+	}
+	if output.ObjectLockConfiguration != nil {
+		objectLockConfigurationptr := output.ObjectLockConfiguration
+		objectLockConfigurationrule := flex.ObjectLockConfigurationGet(objectLockConfigurationptr)
+		if len(objectLockConfigurationrule) > 0 {
+			d.Set("object_lock_configuration", objectLockConfigurationrule)
```

Comment:
Did we define object_lock_configuration in schema?

object_lock is present in schema where did we set the value?